Gizmo overhaul for translate/rotate/scale

kpal81xd opened the pull request:
Continuation of #1367

### Requires
- https://github.com/playcanvas/engine/pull/7960

### What's changed
- Replaces translate/rotate/scale gizmos with ones from engine
- Added Gizmo customisation settings #1431 

### Testing
- Engine `v2.12.4`

Maksims commented:
While new rotation gizmo visually is an upgrade, it's math is changed from the current one, and behaves inconsistently. Depending on steepness and where the circle is grabbed from, the rotation direction will not be towards the mouse, but will move to some offset for some reason.

Here is an image, with blue pointer - where mouse is, and red line - where the rotation vector is pointing (wrong), while it should be pointing as a green line (correct). Current gizmos are doing the correct math currently.

<img width="771" height="548" alt="image" src="https://github.com/user-attachments/assets/cba49364-fc7c-401d-a632-9ce9d97df1a6" />



Video below shows current correct and clear behavior, then shows this PR gizmo with weird offset when rotating:

https://github.com/user-attachments/assets/c60c0c7d-7b69-4dc4-9f84-1badd3517a3a

kpal81xd commented:
@Maksims thanks again for the QA review 😄 

1. & 2.  For flying and rotating while holding the gizmo I have added (surprised this slipped though earlier)

3. The rotation was changed to match Blender to rotating slight angles becomes easier but this makes the guide angle obsolete and requires another guide to show rotation. Scrapped the blender rotation and reverted to the old rotation style

4. Local space gizmo position yep I agree it makes no sense to use average when spaces are different so it uses the last selection space for the gizmos transform

Maksims commented:
Minor one, but nonetheless:

When using ortho camera, the rectangle in translate gizmo, is flipping around - this is actually a current behavior also.
I'm wandering if it has to flip at all.

But other small bit, is when rectangle is flipped, the length of an arrow, is not extended to the middle of the gizmo:

Current:
<img width="333" height="254" alt="image" src="https://github.com/user-attachments/assets/29effa32-5fc4-4b54-875f-03ed66e3190a" />

New:
<img width="223" height="172" alt="image" src="https://github.com/user-attachments/assets/8a9cd6ef-1b9b-4c60-b91e-9ab00543775f" />

Notice that red (X) axis line, starts not from middle when rectangle is flipped, but from far away, making it non-hoverable from the middle.

Also the line thickness feels like less than 1 pixel, making it very thing compared to other lines.

Also rectangle style, I personally like the current one way more, than this solid one.

kpal81xd commented:
> Minor one, but nonetheless:
> 
> When using ortho camera, the rectangle in translate gizmo, is flipping around - this is actually a current behavior also. I'm wandering if it has to flip at all.
> 
> But other small bit, is when rectangle is flipped, the length of an arrow, is not extended to the middle of the gizmo:
> 
> Current: <img alt="image" width="333" height="254" src="https://private-user-images.githubusercontent.com/636263/508145702-29effa32-5fc4-4b54-875f-03ed66e3190a.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjE5MDY0MTEsIm5iZiI6MTc2MTkwNjExMSwicGF0aCI6Ii82MzYyNjMvNTA4MTQ1NzAyLTI5ZWZmYTMyLTVmYzQtNGI1NC04NzVmLTAzZWQ2NmUzMTkwYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMDMxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTAzMVQxMDIxNTFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kNDZkZTgyNjkzY2YxNGQxMmE0ZDNjNmQzMDA2NTkyOTYwNzljOTljZWY5NTE1NWE5MWUzZTk2ZDQ5ODdjM2I4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.sRwFtdXsalsnAJJ7jF8kP6MX83ynSo4SZf8mkVDCb14">
> 
> New: <img alt="image" width="223" height="172" src="https://private-user-images.githubusercontent.com/636263/508145783-8a9cd6ef-1b9b-4c60-b91e-9ab00543775f.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjE5MDY0MTEsIm5iZiI6MTc2MTkwNjExMSwicGF0aCI6Ii82MzYyNjMvNTA4MTQ1NzgzLThhOWNkNmVmLTFiOWItNGM2MC1iOTFlLTlhYjAwNTQzNzc1Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUxMDMxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MTAzMVQxMDIxNTFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kYmNhYzY5MjlmMzgyZWNjM2Q1ODBkNDUwYmJiOTgxZTI2YjQ5MGQwMjI2Y2Y2MjQyMjgyNmM1NGJiZmI1ZTRkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.e5fob9OI-yfk0_k2rGjzmgAjTJGlRr4iBsXD1jJJsTI">
> 
> Notice that red (X) axis line, starts not from middle when rectangle is flipped, but from far away, making it non-hoverable from the middle.
> 
> Also the line thickness feels like less than 1 pixel, making it very thing compared to other lines.
> 
> Also rectangle style, I personally like the current one way more, than this solid one.

@Maksims thanks for testing 🙂 . I will probably do some follow up PRs in the engine down the road to address the inconsistencies in the gizmo design like the line thickness, length and the rectangle opacity but this PR I just want to make sure they function at least as good as the existing ones.
